internal/pkg/skuba/kubernetes/versions_test.go

```diff
@@ -18,13 +18,325 @@
+			clusterVersion: version.MustParseSemantic("1.15.2"),
+			component:      "coredns",
+			expectVersion:  "1.3.1",
+		},
+		{
+			name:           "get pause version when cluster version is 1.15.2",
+			clusterVersion: version.MustParseSemantic("1.15.2"),
+			component:      "pause",
+			expectVersion:  "3.1",
+		},
+		{
+			name:           "get skuba-tooling version when cluster version is 1.15.2",
+			clusterVersion: version.MustParseSemantic("1.15.2"),
+			component:      "tooling",
+			expectVersion:  "0.1.0",
+		},
+		{
+			name:           "get kubelet version when cluster version is 1.15.2",
+			clusterVersion: version.MustParseSemantic("1.15.2"),
+			component:      "kubelet",
+			expectVersion:  "1.15.2",
+		},
+		{
+			name:           "get container version when cluster version is 1.15.2",
+			clusterVersion: version.MustParseSemantic("1.15.2"),
+			component:      "cri-o",
+			expectVersion:  "1.15.0",
+		},
+		{
+			name:           "get hyperkube version when cluster version is 1.15.2",
+			clusterVersion: version.MustParseSemantic("1.15.2"),
+			component:      "hyperkube",
+			expectVersion:  "v1.15.2",
+		},
+		{
+			name:           "get etcd version when cluster version is 1.15.2",
+			clusterVersion: version.MustParseSemantic("1.15.2"),
+			component:      "etcd",
+			expectVersion:  "3.3.11",
+		},
+		{
+			name:           "get coredns version when cluster version is 1.15.2",
+			clusterVersion: version.MustParseSemantic("1.15.2"),
+			component:      "coredns",
+			expectVersion:  "1.3.1",
+		},
+		{
+			name:           "get pause version when cluster version is 1.15.2",
+			clusterVersion: version.MustParseSemantic("1.15.2"),
+			component:      "pause",
+			expectVersion:  "3.1",
+		},
+		{
+			name:           "get skuba-tooling version when cluster version is 1.15.2",
+			clusterVersion: version.MustParseSemantic("1.15.2"),
+			component:      "tooling",
+			expectVersion:  "0.1.0",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			actualVersion := ComponentVersionForClusterVersion(tt.component, tt.clusterVersion)
+			if actualVersion != tt.expectVersion {
+				t.Errorf("returned version (%s) does not match the expected one (%s)", actualVersion, tt.expectVersion)
+				return
+			}
+		})
+	}
+}
+
+func Test_AllComponentContainerImagesForClusterVersion(t *testing.T) {
+	tests := []struct {
+		name           string
+		clusterVersion *version.Version
+	}{
+		{
+			name:           "get all component container images when cluster version is 1.15.2",
+			clusterVersion: version.MustParseSemantic("1.15.2"),
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			actual := AllComponentContainerImagesForClusterVersion(tt.clusterVersion)
+			sort.Slice(actual, func(i int, j int) bool {
+				return actual[i] < actual[j]
+			})
+			expect := []Component{"hyperkube", "etcd", "coredns", "pause", "tooling"}
```

Comment:
If all the test case Component/Addon using existed const, would it become better, WDYT?
https://github.com/SUSE/skuba/blob/master/internal/pkg/skuba/kubernetes/versions.go#L34-L50